Exercise-02/Exercise02/Utilities.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  Utilities.swift
+//  Exercise02
+//
+//  Created by Bilyana Georgieva on 04/02/2019.
+//  Copyright © 2019 Bending Spoons. All rights reserved.
+//
+
+import Foundation
+
+func insertMeasurement(city: String, temperature: Temperature, date: Date, measurements: inout MeasurementsLog) throws -> Measurement {
+  
+  guard isValid(temperature: temperature) else {
+    throw TemperatureError.invalidValue
+  }
+  guard isValid(city: city, date: date, measurements: measurements) else {
+    throw TemperatureError.invalidTime
+  }
+  
+  let measurement = Measurement(city: city, date: date, temperature: temperature)
+  measurements[city] = (measurements[city] ?? []) + [measurement]
+  
+  return measurement
+}
+
+func isValid(temperature: Temperature) -> Bool {
+  return temperature.celsius > -50 && temperature.celsius < 50
+}
+
+func isValid(city: String, date: Date, measurements: MeasurementsLog) -> Bool {
+  let fourHours: TimeInterval = 14_400
```

Comment:
```suggestion
  let fourHours: TimeInterval = 4 * 60 * 60
```
You can use this style when writing time in order to better underline the real duration.